plugins/aws/aws.plugin.zsh

```diff
@@ -92,6 +92,9 @@ else
     # Ubuntu
     elif [[ -e /usr/share/zsh/vendor-completions/_awscli ]]; then
       _aws_zsh_completer_path=/usr/share/zsh/vendor-completions/_awscli
+    # NixOS
+    elif [[ -e "$(dirname "$(dirname $(realpath "$(command -v aws)"))")/share/zsh/site-functions/aws_zsh_completer.sh" ]]; then
+      _aws_zsh_completer_path="$(dirname "$(dirname $(realpath "$(command -v aws)"))")/share/zsh/site-functions/aws_zsh_completer.sh"
```

Comment:
This code runs a fork 3 times at the very least, 6 at the worst. Better use [parameter expansion](http://zsh.sourceforge.net/Doc/Release/Expansion.html#Parameter-Expansion): `${pathvar:h}` for dirname, `${pathvar:P}` for realpath. We can also substitute `command -v aws` for the variable `${commands[aws]}` where `$commands` is a variable holding all the executables in `$PATH`.

Putting it all together:
```suggestion
    elif [[ -e "${commands[aws]:P:h:h}/share/zsh/site-functions/aws_zsh_completer.sh" ]]; then
      _aws_zsh_completer_path="${commands[aws]:P:h:h}/share/zsh/site-functions/aws_zsh_completer.sh"
```

In theory this is the same, test that it works though. BTW, is `realpath` necessary?

Reply:
No, `realpath` seems to be unnecessary. I didn't check that carefully. Thanks for your suggestion. I surely learned some new tricks. I was just trying to edit it on my editor to delete the `:P` part, this PR is been merged too fast. I guess there is no harm in leaving `:P`. Thank you.

Reply:
Oh sorry. Well it doesn't break anything, I'm sure. Might remove it on a later change. Thanks!

Reply:
No problem. Thanks.